File: message/message-java/src/test/java/quan/message/role/RoleInfo.java
package quan.message.role;

import java.util.*;
import java.io.IOException;
import quan.message.*;

/**
 * 角色信息<br/>
 * Created by 自动生成
 */
public class RoleInfo extends Bean {

    //角色id
    private long id;

    private boolean b;

    private short s;

    private int i;

    private float f;

    private double d;

    //角色名
    private String roleName = "";

    private RoleType roleType;

    private byte[] data = new byte[0];

    private ArrayList<Integer> list = new ArrayList<>();

    private HashSet<Integer> set = new HashSet<>();

    private HashMap<Integer, Integer> map = new HashMap<>();

    public RoleInfo() {
    }

    /**
     * 角色id
     */
    public long getId() {
        return id;
    }

    /**
     * 角色id
     */
    public RoleInfo setId(long id) {
        this.id = id;
        return this;
    }

    public boolean getB() {
        return b;
    }

    public RoleInfo setB(boolean b) {
        this.b = b;
        return this;
    }

    public short getS() {
        return s;
    }

    public RoleInfo setS(short s) {
        this.s = s;
        return this;
    }

    public int getI() {
        return i;
    }

    public RoleInfo setI(int i) {
        this.i = i;
        return this;
    }

    public float getF() {
        return f;
    }

    public RoleInfo setF(float f) {
        this.f = f;
        return this;
    }

    public double getD() {
        return d;
    }

    public RoleInfo setD(double d) {
        this.d = d;
        return this;
    }

    /**
     * 角色名
     */
    public String getRoleName() {
        return roleName;
    }

    /**
     * 角色名
     */
    public RoleInfo setRoleName(String roleName) {
        Objects.requireNonNull(roleName);
        this.roleName = roleName;
        return this;
    }

    public RoleType getRoleType() {
        return roleType;
    }

    public RoleInfo setRoleType(RoleType roleType) {
        this.roleType = roleType;
        return this;
    }

    public byte[] getData() {
        return data;
    }

    public RoleInfo setData(byte[] data) {
        Objects.requireNonNull(data);
        this.data = data;
        return this;
    }

    public ArrayList<Integer> getList() {
        return list;
    }

    public HashSet<Integer> getSet() {
        return set;
    }

    public HashMap<Integer, Integer> getMap() {
        return map;
    }

    @Override
    public void encode(Buffer buffer) throws IOException {
        super.encode(buffer);

        buffer.writeLong(id);
        buffer.writeBool(b);
        buffer.writeShort(s);
        buffer.writeInt(i);
        buffer.writeFloat(f);
        buffer.writeDouble(d);
        buffer.writeString(roleName);

        if(roleType != null) {
            buffer.writeInt(roleType.getValue());
        }else {
            buffer.writeInt(0);
        }

        buffer.writeBytes(data);

        buffer.writeInt(list.size());
        for (int $list$Value : list) {
            buffer.writeInt($list$Value);
        }

        buffer.writeInt(set.size());
        for (int $set$Value : set) {
            buffer.writeInt($set$Value);
        }

        buffer.writeInt(map.size());
        for (int $map$Key : map.keySet()) {
            buffer.writeInt($map$Key);
            buffer.writeInt(map.get($map$Key));
        }
    }

    @Override
    public void decode(Buffer buffer) throws IOException {
        super.decode(buffer);

        id = buffer.readLong();
        b = buffer.readBool();
        s = buffer.readShort();
        i = buffer.readInt();
        f = buffer.readFloat();
        d = buffer.readDouble();
        roleName = buffer.readString();
        roleType = RoleType.valueOf(buffer.readInt());
        data = buffer.readBytes();

        int $list$Size = buffer.readInt();
        for (int i = 0; i < $list$Size; i++) {
            list.add(buffer.readInt());
        }

        int $set$Size = buffer.readInt();
        for (int i = 0; i < $set$Size; i++) {
            set.add(buffer.readInt());
        }

        int $map$Size = buffer.readInt();
        for (int i = 0; i < $map$Size; i++) {
            map.put(buffer.readInt(), buffer.readInt());
        }
    }

    @Override
    public String toString() {
        return "RoleInfo{" +
                "id=" + id +
                ",b=" + b +
                ",s=" + s +
                ",i=" + i +
                ",f=" + f +
                ",d=" + d +
                ",roleName='" + roleName + '\'' +
                ",roleType=" + roleType +
                ",data=" + Arrays.toString(data) +
                ",list=" + list +
                ",set=" + set +
                ",map=" + map +
                '}';

    }

}

File: message/MessageCS/Test/Role/RoleInfo.cs
using System;
using System.Collections.Generic;
using MessageCS;
using Buffer = MessageCS.Buffer;

namespace MessageCS.Test.Role
{
	/// <summary>
	/// 角色信息<br/>
	/// Created by 自动生成
	/// </summary>
    public class RoleInfo : Bean
    {
        /// <summary>
		/// 角色id
		/// </summary>
		public long Id { get; set; }

		public bool B { get; set; }

		public short S { get; set; }

		public int I { get; set; }

		public float F { get; set; }

		public double D { get; set; }

		private string _roleName = "";

        /// <summary>
		/// 角色名
		/// </summary>
		public string RoleName
		{
	    	get => _roleName;
	    	set => _roleName = value ?? throw new NullReferenceException();
		}

		public RoleType RoleType { get; set; }

		private byte[] _data = new byte[0];

		public byte[] Data
		{
            get => _data;
            set => _data = value ?? throw new NullReferenceException();
		}

		public List<int> List { get; } = new List<int>();

		public HashSet<int> Set { get; } = new HashSet<int>();

		public Dictionary<int, int> Map { get; } = new Dictionary<int, int>();



		public override void Encode(Buffer buffer)
		{
	    	base.Encode(buffer);

		    buffer.WriteLong(Id);
		    buffer.WriteBool(B);
		    buffer.WriteShort(S);
		    buffer.WriteInt(I);
		    buffer.WriteFloat(F);
		    buffer.WriteDouble(D);
		    buffer.WriteString(RoleName);
			buffer.WriteInt((int)RoleType);
		    buffer.WriteBytes(Data);

		    buffer.WriteInt(List.Count);
		    foreach (var listValue in List) {
			    buffer.WriteInt(listValue);
		    }

		    buffer.WriteInt(Set.Count);
		    foreach (var setValue in Set) {
			    buffer.WriteInt(setValue);
		    }

		    buffer.WriteInt(Map.Count);
		    foreach (var mapKey in Map.Keys) {
		        buffer.WriteInt(mapKey);
			    buffer.WriteInt(Map[mapKey]);
		    }
		}

		public override void Decode(Buffer buffer)
		{
	    	base.Decode(buffer);

		    Id = buffer.ReadLong();
		    B = buffer.ReadBool();
		    S = buffer.ReadShort();
		    I = buffer.ReadInt();
		    F = buffer.ReadFloat();
		    D = buffer.ReadDouble();
		    RoleName = buffer.ReadString();
		    RoleType = (RoleType)buffer.ReadInt();
		    Data = buffer.ReadBytes();

		    var listSize = buffer.ReadInt();
		    for (var i = 0; i < listSize; i++) {
			    List.Add(buffer.ReadInt());
		    }

		    var setSize = buffer.ReadInt();
		    for (var i = 0; i < setSize; i++) {
			    Set.Add(buffer.ReadInt());
		    }

		    var mapSize = buffer.ReadInt();
		    for (var i = 0; i < mapSize; i++) {
			    Map.Add(buffer.ReadInt(), buffer.ReadInt());
		    }
		}

		public override string ToString()
		{
			return "RoleInfo{" +
					"id=" + Id +
					",b=" + B +
					",s=" + S +
					",i=" + I +
					",f=" + F +
					",d=" + D +
					",roleName='" + RoleName + '\'' +
					",roleType=" + RoleType +
					",data=" + Convert.ToBase64String(Data) +
					",list=" + ToString(List) +
					",set=" + ToString(Set) +
					",map=" + ToString(Map) +
					'}';
		}
    }
}